### PR TITLE
Correctly resolve trust_remote_code=None for AutoTokenizer

### PR DESCRIPTION
**Pull request by Rocketknight1**
If `trust_remote_code` is left at the default `None` in `AutoTokenizer.from_pretrained()`, an error is thrown if you try to load a repo that requires remote code, rather than the correct dialog box being displayed. This PR fixes that issue.

### CODE REVIEW

File: src/transformers/models/auto/tokenization_auto.py [hunk 1]
```diff
@@ -769,7 +769,13 @@ def from_pretrained(cls, pretrained_model_name_or_path, *inputs, **kwargs):
...
 
         has_remote_code = tokenizer_auto_map is not None
-        has_local_code = config_tokenizer_class is not None or type(config) in TOKENIZER_MAPPING
+        has_local_code = type(config) in TOKENIZER_MAPPING or (
+            config_tokenizer_class is not None
+            and (
+                tokenizer_class_from_name(config_tokenizer_class) is not None
+                or tokenizer_class_from_name(config_tokenizer_class + "Fast") is not None
+            )
+        )
```
Comment (Rocketknight1) [inline src/transformers/models/auto/tokenization_auto.py:778]: Comment for the reviewer here: Before this change, models with custom tokenizer code would usually set `config_tokenizer_class`, but this class would not be one of the recognized library classes. As a result, when `trust_remote_code=None` for those repos, we would incorrectly infer `has_local_code=True`, and then this code block would continue before hitting the `Tokenizer class does not exist or is not currently imported` error clause below.

This change ensures we only infer `has_local_code=True` when `config_tokenizer_class` is set **and** we actually recognize that class as one of the library classes.

**Review by ArthurZucker — COMMENTED** [review]
Thanks it makes sense. I think there was an issue with this in a colab where trust remote code was not passed properly but the PR got closes I think. 
Anyway, LGTM we should probably add a small dummy test no?

**Comment by Rocketknight1** [comment]
I tried, but I couldn't figure out a way to test it, and I don't think the previous test authors could either! The test would need to check that `input()` is called, but we can't really check that - we can only check for the error that's raised when it times out. This is what happens in [the existing tests](https://github.com/huggingface/transformers/blob/main/tests/models/auto/test_tokenization_auto.py#L306), and that's why the tests didn't catch the problem - the error is raised without ever calling `input()`, and the test only checks for the error, so it passed.

**Review by ArthurZucker — APPROVED** [review]
Alright good to go then!

**Pull request merged**